Cryptomator/Purchase/PurchaseCoordinator.swift

```diff
@@ -24,18 +24,20 @@ class PurchaseCoordinator: Coordinator {
 	func start() {
 		let purchaseViewController = PurchaseViewController(viewModel: PurchaseViewModel())
 		purchaseViewController.coordinator = self
+		purchaseViewController.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(doneButtonTapped))
 		navigationController.pushViewController(purchaseViewController, animated: true)
 	}
 
-	func showUpgrade() {
+	func showUpgrade(onAlertDismiss execute: (() -> Void)? = nil) {
 		if UpgradeChecker.shared.isEligibleForUpgrade() {
-			let child = getUpgradeCoordinator()
-			childCoordinators.append(child) // TODO: remove missing?
-			child.start()
+			let upgradeViewController = IAPViewController(viewModel: UpgradeViewModel())
+			upgradeViewController.coordinator = getUpgradeCoordinator()
+			upgradeViewController.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(doneButtonTapped))
+			navigationController.pushViewController(upgradeViewController, animated: true)
 		} else if UIApplication.shared.canOpenURL(UpgradeChecker.upgradeURL) {
 			UIApplication.shared.open(UpgradeChecker.upgradeURL)
 		} else {
-			showUpgradeFailedAlert()
+			showUpgradeFailedAlert(onAlertDismiss: execute)
 		}
 	}
```

Comment:
It's not really part of this particular code but I noticed that `couldBeEligibleForUpgrade()` of `UpgradeChecker` is unused. I guess it can be deleted?